xdsl/dialects/stream.py

```diff
@@ -79,66 +60,20 @@ def constr(
 class ReadableStreamType(Generic[_StreamTypeElement], StreamType[_StreamTypeElement]):
     name = "stream.readable"
 
-    @overload
-    @staticmethod
-    def constr(
-        *,
-        element_type: None = None,
-    ) -> BaseAttr[ReadableStreamType[Attribute]]: ...
 
-    @overload
-    @staticmethod
-    def constr(
-        *,
-        element_type: GenericAttrConstraint[_StreamTypeElementConstrT],
-    ) -> ParamAttrConstraint[ReadableStreamType[_StreamTypeElementConstrT]]: ...
-
-    @staticmethod
-    def constr(
-        *,
-        element_type: GenericAttrConstraint[_StreamTypeElementConstrT] | None = None,
-    ) -> (
-        BaseAttr[ReadableStreamType[Attribute]]
-        | ParamAttrConstraint[ReadableStreamType[_StreamTypeElementConstrT]]
-    ):
-        if element_type is None:
-            return BaseAttr[ReadableStreamType[Attribute]](ReadableStreamType)
-        return ParamAttrConstraint[ReadableStreamType[_StreamTypeElementConstrT]](
-            ReadableStreamType, (element_type,)
-        )
+AnyReadableStreamTypeConstr = BaseAttr[ReadableStreamType[Attribute]](
+    ReadableStreamType
+)
 
 
 @irdl_attr_definition
 class WritableStreamType(Generic[_StreamTypeElement], StreamType[_StreamTypeElement]):
     name = "stream.writable"
 
-    @overload
-    @staticmethod
-    def constr(
-        *,
-        element_type: None = None,
-    ) -> BaseAttr[WritableStreamType[Attribute]]: ...
 
-    @overload
-    @staticmethod
-    def constr(
-        *,
-        element_type: GenericAttrConstraint[_StreamTypeElementConstrT],
-    ) -> ParamAttrConstraint[WritableStreamType[_StreamTypeElementConstrT]]: ...
-
-    @staticmethod
-    def constr(
-        *,
-        element_type: GenericAttrConstraint[_StreamTypeElementConstrT] | None = None,
-    ) -> (
-        BaseAttr[WritableStreamType[Attribute]]
-        | ParamAttrConstraint[WritableStreamType[_StreamTypeElementConstrT]]
-    ):
-        if element_type is None:
-            return BaseAttr[WritableStreamType[Attribute]](WritableStreamType)
-        return ParamAttrConstraint[WritableStreamType[_StreamTypeElementConstrT]](
-            WritableStreamType, (element_type,)
-        )
+AnyWritableStreamTypeConstr = BaseAttr[WritableStreamType[Attribute]](
+    WritableStreamType
+)
```

Comment:
I have a feeling that this will be the pattern to use until we have defaults for TypeVars